tests/entities/test_experiment.py

```diff
@@ -1,67 +1,91 @@
-        assert exp.creation_time == creation_time
-        assert exp.last_update_time == last_update_time
-
-    def test_creation_and_hydration(self):
-        exp_id = str(random_int())
-        name = "exp_%d_%d" % (random_int(), random_int())
-        lifecycle_stage = LifecycleStage.ACTIVE
-        location = random_file(".json")
-        creation_time = get_current_time_millis()
-        last_update_time = get_current_time_millis()
-
-        exp = Experiment(
-            exp_id,
-            name,
-            location,
-            lifecycle_stage,
-            creation_time=creation_time,
-            last_update_time=last_update_time,
-        )
-        self._check(exp, exp_id, name, location, lifecycle_stage, creation_time, last_update_time)
-
-        as_dict = {
-            "experiment_id": exp_id,
-            "name": name,
-            "artifact_location": location,
-            "lifecycle_stage": lifecycle_stage,
-            "tags": {},
-            "creation_time": creation_time,
-            "last_update_time": last_update_time,
-        }
-        assert dict(exp) == as_dict
-
-        proto = exp.to_proto()
-        exp2 = Experiment.from_proto(proto)
-        self._check(exp2, exp_id, name, location, lifecycle_stage, creation_time, last_update_time)
-
-        exp3 = Experiment.from_dictionary(as_dict)
-        self._check(exp3, exp_id, name, location, lifecycle_stage, creation_time, last_update_time)
-
-    def test_string_repr(self):
-        exp = Experiment(
-            experiment_id=0,
-            name="myname",
-            artifact_location="hi",
-            lifecycle_stage=LifecycleStage.ACTIVE,
-            creation_time=1662004217511,
-            last_update_time=1662004217511,
-        )
-        assert (
-            str(exp)
-            == "<Experiment: artifact_location='hi', creation_time=1662004217511, experiment_id=0, "
-            "last_update_time=1662004217511, lifecycle_stage='active', name='myname', tags={}>"
-        )
+def _check(exp, exp_id, name, location, lifecyle_stage, creation_time, last_update_time):
+    assert isinstance(exp, Experiment)
+    assert exp.experiment_id == exp_id
+    assert exp.name == name
+    assert exp.artifact_location == location
+    assert exp.lifecycle_stage == lifecyle_stage
+    assert exp.creation_time == creation_time
+    assert exp.last_update_time == last_update_time
+
+
+@pytest.fixture
+def exp_id(scope="module"):
+    yield str(random_int())
+
+
+@pytest.fixture
+def name(scope="module"):
+    yield "exp_%d_%d" % (random_int(), random_int())
+
+
+@pytest.fixture
+def lifecycle_stage(scope="module"):
+    yield LifecycleStage.ACTIVE
+
+
+@pytest.fixture
+def location(scope="module"):
+    yield random_file(".json")
+
+
+@pytest.fixture
+def creation_time(scope="module"):
+    yield get_current_time_millis()
+
+
+@pytest.fixture
+def last_update_time(scope="module"):
+    yield get_current_time_millis()
```

Comment:
Can we directly use `get_current_time_millis`?

Reply:
solved in https://github.com/wkrt7/mlflow/commit/3a76eb2e91490a13ea70ca338c5a84dd491f8c26